environs/imagedownloads/simplestreams.go

```diff
@@ -99,18 +99,21 @@ func Fetch(
 			ValueTemplate: Metadata{},
 		},
 	}
-	items, resolveInfo, err := simplestreams.GetMetadata(src, params)
+	results, err := simplestreams.GetMetadata(src, params)
```

Comment:
GetMetadata should return NotFound if len(results) == 0 
This simplifies all of the call sites to not require a separate error check

Reply:
Wilco.